service/src/networking/SignalCommunicator.ts

```diff
@@ -12,11 +12,13 @@ class SignalCommunicator {
                     delete this.#connections[request.clientId]
                 }
             })
-            this.#onConnectionCallbacks.forEach(cb => {cb(cc)})
+            for (const cb of this.#onConnectionCallbacks) {
+                await cb(cc)
+            }
```

Comment:
Can we do `await Promise.all(this.#onConnectionCallbacks.forEach(cb => cb(cc)))`? I think that should work to call them all in parallel, but not sure how to test it...

Reply:
I agree this would be better, but I think you mean .map rather than .forEach

I think running the service and trying it out would be the best way to test.